taskcluster/android_taskgraph/transforms/gradle_optimization.py

```diff
@@ -9,27 +9,27 @@
 @transforms.add
 def add_components_optimization(config, tasks):
     for task in tasks:
-        if _is_task_related_to_android_components(task):
-            build_type = task["attributes"]["build-type"]
-            if build_type not in ("nightly", "release"):
-                optimization = task.setdefault("optimization", {})
-                skip_unless_changed = optimization.setdefault("skip-unless-changed", [])
-                skip_unless_changed.extend([
-                    "android-components/build.gradle",
-                    "android-components/settings.gradle",
-                    "android-components/buildSrc.*",
-                    "android-components/gradle.properties",
-                    "android-components/gradle/wrapper/gradle-wrapper.properties",
-                    "android-components/plugins/dependencies/**",
-                ])
+        attributes = task.get("attributes", {})
+        build_type = attributes.get("build-type", "")
+        release_type = attributes.get("release-type", "")
+        # We want to optimize away tasks on all components and APKs as long as they're
```

Comment:
I'm having trouble parsing this comment, e.g. who's "they" here, is it tasks, components, or apks?
What's "a general change in a-c"?